### PR TITLE
Support for GitLab 9.1, 9.2 and 9.3.

### PR DESCRIPTION
**Pull request by bfabio**

### CODE REVIEW

File: tasks/configure_gitlab_ce.yml [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
+  apt_repository:
+    repo: '{{ gitlab__yarn_apt_repo }}'
+    state: 'present'
+    update_cache: True
+  when: gitlab_version | version_compare("8.17", operator="ge", strict=True)
+
+- name: Install Yarn
+  apt:
+    name: 'yarn'
+    state: 'latest'
```
Comment (drybjed) [inline tasks/configure_gitlab_ce.yml]: This should be set to `present` so that the role does not upgrade the packages on a subsequent run.

**Comment by drybjed** [comment]
@bfabio Nice work. However, @cultcom is also working on a parallel PR https://github.com/debops/ansible-gitlab/pull/81 which adds support for Debian Stretch. The Travis-CI test at the moment cannot handle both PRs because you are changing the playbook and removing a variable (which is fine, BTW, no issue there).

How about this - let's do the Stretch changes first because the current test playbook works with that PR, then I'll change the playbook and we do the 9.x PR next?

**Comment by bfabio** [comment]
@drybjed Sound good.

**Comment by cultcom** [comment]
@bfabio Hi, I already had something similar in my queue, look at my changes here:https://github.com/cultcom/ansible-gitlab/tree/cultcom-9.x

Maybe you can adopt the gitaly versions and the service file. The systemd service file still needs some attendance I think.

**Comment by drybjed** [comment]
@bfabio, can you rebase this PR on the recent changes merged in the `master` branch? I'll update the Travis test playbook to not include the removed variable.

File: defaults/main.yml [hunk 2]
```diff
@@ -460,6 +461,23 @@ gitlab_shell_ssh_port: '22'
...
+# .. envvar:: gitlab__yarn_apt_key_id [[[
+#
+# The OpenPGP key used to sign Yarn repository.
+gitlab__yarn_apt_key_id: '72ECF46A56B4AD39C907BBB71646B01B86E50310'
+
+                                                                   # ]]]
+# .. env/var:: gitlab__yarn_apt_repo [[[
+#
+# APT repository URL to Yarn repository.
+gitlab__yarn_apt_repo: 'deb https://dl.yarnpkg.com/debian/ {{ ansible_distribution_release }} main'
```
Comment (drybjed) [inline defaults/main.yml]: The string needs to use `stable` and not a specific OS release to work correctly.

**Comment by cultcom** [comment]
@bfabio what are 'libsqlite3-dev' and 'python-mysqldb' meant for?

the later one should be included by debops.mariadb_server or am I wrong here?

**Comment by cultcom** [comment]
@bfabio and the "gitconfig" template does not need a version check for > 9.x. The writeback was first mentioned here https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/update/8.11-to-8.12.md

**Comment by bfabio** [comment]
@cultcom `python-mysqldb` should not be required, I misinterpreted the output from the CI build. 

`libsqlite3-dev` should be required by `bundle install`, I think it can be removed using `gitlab_ce_bundle_install_without`. Let's see what Travis has to say about that.

I removed the conditional in `gitconfig.j2`, thanks for the hint.

File: defaults/main.yml (hunk 2 above)
Comment (bfabio) [inline defaults/main.yml]: Thanks, fixed.

**Comment by bfabio** [comment]
@cultcom Also, I'm going to merge your changes if that's ok with you.

**Comment by drybjed** [comment]
@bfabio Forgot `apt_key` task, I think?

**Comment by bfabio** [comment]
@drybjed Yup :)

**Comment by cultcom** [comment]
@bfabio I just pushed a cleanup for the gitaly service file, have a look at it as well please. Your are welcome to integrate the useful stuff from my branch ;-)

**Comment by drybjed** [comment]
Guys, don't forget to add yourselves to the `COPYRIGHT` file.

**Comment by drybjed** [comment]
@bfabio, while you are working on this PR, is it OK for me to merge https://github.com/debops/ansible-gitlab/pull/86 which has mostly cosmetic changes? You will need to rebase your PR afterwards.

**Comment by bfabio** [comment]
@drybjed, sure, go ahead.

**Comment by drybjed** [comment]
@bfabio The PR has been merged, you need to rebase this one.

**Comment by drybjed** [comment]
@bfabio Anything else planned for the PR? Let me know when it's ready to merge.

**Comment by drybjed** [comment]
@bfabio Another PR merged, can you fix the conflicts and rebase once again? I guess this PR can be merged next?

**Comment by bfabio** [comment]
@drybjed I would like to add @cultcom's work and install `gitaly` as well, because it turns out gitlab-shell doesn't handle its absence, even when `gitaly` is explicitly disabled in the configuration file.

**Comment by cultcom** [comment]
@drybjed @bfabio I am not sure but until 9.2 the documentation mentioned gitaly as optional. The 9.3 update guide is missing that part but does not explicitly mark it mandatory. On the other hand the example config changed to use it by default.

**Comment by alinalexandru** [comment]
Is there anything left to do until it will be merged?
I can help, as I want to be able to install version 9.3 via ansible.

**Comment by cultcom** [comment]
@alinalexandru I think the gitaly installation part is still missing if you like you may add it. Unfortunately I am currently out of time for it.

**Comment by alinalexandru** [comment]
Don't know what is gitaly. I have to learn about it

**Comment by alinalexandru** [comment]
This steps fo I have to follow ? 
https://docs.gitlab.com/ce/install/installation.html#install-gitaly

**Comment by drybjed** [comment]
@alinalexandru Yes, sounds like it. Keep in mind that `debops.gitlab` uses different home directory path. You can check how IIRC `gitlab-workhorse` was installed to see how the role handled Go applications.

File: defaults/main.yml [hunk 3]
```diff
@@ -679,6 +730,27 @@ gitlab_pages_dest: '{{ gitlab_src_path + "/" + gitlab_pages_repo.split("://")[1]
 gitlab_pages_checkout: '{{ gitlab_app_root_path + "/gitlab-pages" }}'
                                                                    # ]]]
                                                                    # ]]]
+# Gitaly source code [[[
+# --------------------------------
+
+# .. envvar:: gitaly_repo [[[
+#
+# URL of the Gitaly repository.
+gitaly_repo: 'https://gitlab.com/gitlab-org/gitaly.git'
```
Comment (drybjed) [inline defaults/main.yml]: This and other `gitaly_` variables need to use the `gitlab__` variable namespace. Please rename them to `gitlab__gitaly_*`.

File: defaults/main.yml [hunk 4]
```diff
@@ -679,6 +730,27 @@ gitlab_pages_dest: '{{ gitlab_src_path + "/" + gitlab_pages_repo.split("://")[1]
 gitlab_pages_checkout: '{{ gitlab_app_root_path + "/gitlab-pages" }}'
                                                                    # ]]]
                                                                    # ]]]
+# Gitaly source code [[[
+# --------------------------------
+
+# .. envvar:: gitaly_repo [[[
```
Comment (drybjed) [inline defaults/main.yml]: The headers also need to be renamed, otherwise references in the documentation will be broken.

Comment (bfabio) [inline defaults/main.yml]: Oops, you are right.

**Comment by bfabio** [comment]
This should be all.

I didn't test this in production though, so testers are welcome.

**Comment by drybjed** [comment]
@bfabio Great, thanks for your work!

@alinalexandru Any chance for you to test this PR before merging?

**Comment by alinalexandru** [comment]
Only next week.
BTW. are the runners updated?

**Comment by drybjed** [comment]
@alinalexandru Do you mean `debops.gitlab_runner`? There are some pending PRs, I'll try to look them over during the weekend.

**Comment by alinalexandru** [comment]
meanwhile 9.4 was released 😄

**Comment by drybjed** [comment]
@alinalexandru I'm testing the upgrade path of this PR on one of my hosts, if it passes, I'm going to merge it in about 1-2 hours.

**Comment by alinalexandru** [comment]
@drybjed Do you think is difficult to add support for 9.4?

**Comment by drybjed** [comment]
From the [upgrade instructions](https://gitlab.com/gitlab-org/gitlab-ce/blob/9-4-stable/doc/update/patch_versions.md) it looks like there are no major changes. Just adding the correct versions to the version map should be enough. Want to try it?

**Comment by alinalexandru** [comment]
I don't have time now, but I will try when I will put gitlab via ansible.

**Pull request merged**